DataReport/collectBucketMetricsAndUpdateBucketCapacityInfo.js

```diff
@@ -0,0 +1,126 @@
+const async = require('async');
+const { MaxParallelLimit } = require('../utils/constants');
+
+function isSOSCapacityInfoEnabled(bucketInfo) {
+    /**
+     * a valid SOS CapacityInfo Enabled bucketMD looks like
+     * {
+     *     _capabilities: {
+     *         VeeamSOSApi: {
+     *             SystemInfo: {
+     *                 ...
+     *                 ProtocolCapabilities: {
+     *                     CapacityInfo: true,
+     *                     ...
+     *                 },
+     *             ...
+     *             },
+     *             CapacityInfo: {
+     *                 ...
+     *             },
+     *         }
+     *     },
+     * }
+     */
+    return !!(bucketInfo.getCapabilities() // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities.CapacityInfo === true // should be true
+        && bucketInfo.getCapabilities().VeeamSOSApi.CapacityInfo); // should exist
+}
+
+function isValidBucketStorageMetrics(bucketMetric) {
+    return bucketMetric
+    && bucketMetric.usedCapacity
+        && typeof bucketMetric.usedCapacity.current === 'number'
+        && typeof bucketMetric.usedCapacity.nonCurrent === 'number'
+    && bucketMetric.usedCapacity.current > -1
+    && bucketMetric.usedCapacity.nonCurrent > -1;
+}
+
+function isValidCapacityValue(capacity) {
+    return (typeof capacity === 'number' && capacity >= 0);
```

Comment:
Just wondering if we shoudl reject floating numbers? They will be accepted here if the user puts something like `123456.789`. But it's a minor concern.

Reply:
We should round off.
what do you think?

Reply:
discussed further over slack, as we are always dealing in bytes, rejecting floats should be okay.